### PR TITLE
Make sure that roles order will be refreshed for dns-servers

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

**Comment by mjura** [comment]
I checked this patch localy, works properly

**Comment by tboerger** [comment]
LGTM

**Comment by rhafer** [comment]
Do we need to do somethign with the nodes as well in the downgrade case? I like re-inserting in the old order?

**Comment by mjura** [comment]
> Do we need to do somethign with the nodes as well in the downgrade case? I like re-inserting in the old order?

No, it will be not needed, new order will work also for old version.

**Comment by rhafer** [comment]
Ok, I guess then it's fine. +1

**Comment by ss7pro** [comment]
LGTM

**Comment by bmwiedemann** [comment]
+1

File: chef/data_bags/crowbar/migrate/dns/003_add_element_run_list_order.rb [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+
+  # Make sure that all dns-servers will refresh roles order for run_list
+  nodes = NodeObject.find('roles:dns-server')
+  nodes.each do |node|
+    node.delete_from_run_list('dns-client')
+    node.add_to_run_list('dns-client',
+                         td['element_run_list_order']['dns-client'],
+                         td['element_states']['dns-client'])
+    node.save
+  end
```
Comment (vuntz) [inline chef/data_bags/crowbar/migrate/dns/003_add_element_run_list_order.rb]: There are two bugs here:

 - if a dns-server node doesn't have the dns-client role (okay, unlikely)
 - we need to do that for all dns-client nodes as they might get a dns-server role later on

**Pull request merged**